programs/uxd/src/utils/calculate_depositories_redeem_redeemable_amount.rs

```diff
@@ -0,0 +1,203 @@
+use anchor_lang::prelude::Result;
+use anchor_lang::require;
+
+use crate::error::UxdError;
+use crate::utils::calculate_depositories_sum_value;
+
+use super::checked_convert_u128_to_u64;
+use super::compute_amount_less_fraction_floor;
+
+pub struct DepositoriesRedeemRedeemableAmount {
```

Comment:
Possibly `DepositoriesRedeemRedeemableAmount` -> `DepositoriesRedeemableAmount`?

Reply:
I named it not just "RedeemableAmount" so that we can know the difference between:
- depositories_target_redeemable_amount
- depositories_redeem_redeemable_amount

the "redeem-redeemable" is a bit redundant, i dont have too much inspiration for this naming, definitly would love to find a better name lol

Reply:
But the redeem is superfluous here I still think, these two variable already have a difference which is the `target`?

depositories_target_redeemable_amount
depositories_redeemable_amount

Reply:
Sounds good, ok, doing on next PR to not waste this review+approval